### ISSUE TITLE
why sucrase is not a parallel?

### ISSUE THREAD

**Issue by a-x-** [alangpierce/sucrase#459]
it's easy to add thread pool and speed it up twice or more.

**Comment by alangpierce** [comment]
Agreed that parallelization has a lot of potential here! For Sucrase specifically, though, I'd prefer to call it out of scope for the core Sucrase library. Sucrase's `transform` is a plain JavaScript function that synchronously gives you a string of the transformed code, and that means that it can run in any environment (in the browser, in node, invoked by webpack or a CLI or a custom build system, etc). Nothing stops you from invoking Sucrase's `transform` function multiple times in parallel, and in fact I'd even encourage it!

As one example, if you're using Sucrase with webpack, you can use happypack ( https://github.com/amireh/happypack ) to run it across multiple worker processes. I do this at my work, though in my case I just used one background process rather than multiple. Sucrase only takes a second or two to run on the whole codebase, whereas webpack takes about 20 seconds to process the Sucrase output, so Sucrase is no longer the bottleneck there.

I think require hooks in general are hard to make parallel, which is why `sucrase/register` doesn't make any attempt at that right now.

Seems pretty reasonable to make the `sucrase` command line tool run in parallel, if that's what you had in mind. In general there's a lot of potential improvement for that tool.

**Comment by a-x-** [comment]
> Seems pretty reasonable to make the sucrase command line tool run in parallel, if that's what you had in mind. In general there's a lot of potential improvement for that tool.

yeah! Or you can allow to run sucrase for a single file

I use something like this for postcss in my npm scripts:
```sh
time ls src/**/*.css | parallel  --jobs 8 --shell postcss '--config .postcssrc.js -o $(echo {} | sed '"'s/src/build/'"')  {}'
```

**Comment by a-x-** [comment]
thread-loader seems should be preferred over the happypack

https://github.com/amireh/happypack#is-it-necessary-for-webpack-4
